packages/analytics-js/src/services/ErrorHandler/ErrorHandler.ts

```diff
@@ -76,25 +81,38 @@ class ErrorHandler implements IErrorHandler {
     context = '',
     customMessage = '',
     shouldAlwaysThrow = false,
-    errorType = 'handledException',
+    errorType = ErrorType.HANDLEDEXCEPTION,
   ) {
-    // Error handling is already implemented in processError method
-    let errorMessage = processError(error);
-
-    // If no error message after we normalize, then we swallow/ignore the errors
-    if (!errorMessage) {
-      return;
-    }
+    let normalizedError;
+    let errorMessage;
+    if (errorType === ErrorType.HANDLEDEXCEPTION) {
+      errorMessage = processError(error);
+
+      // If no error message after we normalize, then we swallow/ignore the errors
+      if (!errorMessage) {
+        return;
+      }
 
-    errorMessage = removeDoubleSpaces(
-      `${context}${LOG_CONTEXT_SEPARATOR}${customMessage} ${errorMessage}`,
-    );
+      errorMessage = removeDoubleSpaces(
+        `${context}${LOG_CONTEXT_SEPARATOR}${customMessage} ${errorMessage}`,
+      );
 
-    let normalizedError = new Error(errorMessage);
-    if (isTypeOfError(error)) {
-      normalizedError = Object.create(error, {
-        message: { value: errorMessage },
-      });
+      normalizedError = new Error(errorMessage);
+      if (isTypeOfError(error)) {
+        normalizedError = Object.create(error, {
+          message: { value: errorMessage },
+        });
+      }
+    } else {
+      errorMessage = normalizeErrorMessageForUnhandledError(error);
+      if (!errorMessage) {
+        return;
+      }
+      normalizedError = errorMessage.startsWith('Error in loading a third-party script')
+        ? Object.create(error as Event, {
```

Comment:
This error cloning can be done within the normalization logic itself, where the final returned error is the normalized error we need, and it'll also contain the error message.

Reply:
Addressed